sub_accounts/client_sub_account_create.go

```diff
@@ -104,6 +104,10 @@ func validateCreateSubAccount(createSubAccount CreateOrUpdateSubAccount) error {
 			return fmt.Errorf("SnapSearchRetentionDays cannot be set if retentionDays is less than 4")
 		}
 	}
-	
+
+	if createSubAccount.SoftLimitGB != nil && *createSubAccount.SoftLimitGB <= 0 {
```

Comment:
I think the validation should also cover `Flexible` since this feature should work only for on demand accounts